spec/features/checkout_spec.rb

```diff
@@ -69,7 +69,25 @@
         expect(page).to have_content(variant.product.name)
       end
     end
+  end
 
+  context "backend order shipments UI", js: true do
+    context 'when the product and parts are backorderable' do
+      it_behaves_like 'purchasable assembly'
+    end
+
+    context 'when the product and parts are not backorderable' do
+      context 'when there is enough quantity available' do
+        before do
+          Spree::StockItem.update_all backorderable: false
+          variant.stock_items.each { |stock_item| stock_item.set_count_on_hand 2 }
+          other_variant.stock_items.each { |stock_item| stock_item.set_count_on_hand 1 }
+          product.stock_items.each { |stock_item| stock_item.set_count_on_hand 1 }
```

Comment:
I'm not sure why this line is necessary. Without it, the assembly is not purchasable because `out of stock`.
This puzzles me a bit, as this product should be only a "container" for its two parts.

Reply:
I'm not sure about that. The Readme states that

> Spree will create and track inventory units for its parts rather than for the product itself

and that makes sense: if a product is a dummy wrapper (in regards to stock) then the only way to manipulate product availability should be to increase/decrease its parts stock.

To be more explicit, if a bundle is made of 2 variants and both of them are available, then the product should be available. If one of them is out of stock, then the product should be out of stock... at least that's what I'd expect after the Readme.

Reply:
I think it's a legit doubt, I expect the wrapper product count on hand to be equal to the minimum value of the count on hand of its parts but maybe this is left to each store's implementation?